extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -168,6 +169,21 @@ private String guessDriver(Optional<JdbcDataSourceBuildItem> jdbcDataSource) {
                     .add(new ReflectiveClassBuildItem(true, true, "io.quarkus.quartz.runtime.QuartzScheduler$InvokerJob"));
             reflectiveClasses
                     .add(new ReflectiveClassBuildItem(true, false, QuarkusQuartzConnectionPoolProvider.class.getName()));
+        } else if (config.storeType.isCustom()) {
+            Optional<QuartzExtensionPointConfig> customStore = config.customStore();
+            if (!customStore.isPresent()) {
+                throw new ConfigurationException("Custom job store class is not configured properly");
+            }
+            try {
+                if (!JobStore.class.isAssignableFrom(
+                        Class.forName(customStore.get().clazz, false, Thread.currentThread().getContextClassLoader()))) {
+                    throw new IllegalArgumentException(
+                            String.format("%s does not implements %s", customStore.get().clazz, JobStore.class));
```

Comment:
Maybe a configuration exception will be more appropriate here,just like: https://github.com/quarkusio/quarkus/blob/e82004d5e20184ac6432bcc7b9358b201b9107f1/extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java#L101